src/libs/error.js

```diff
@@ -12,20 +12,35 @@ export const reportError = async (title, obj) => {
   }
 }
 
-// Transforms an async function so that it catches and reports errors using the provided text
-export const withErrorReporting = _.curry((title, fn) => async (...args) => {
+/**
+ * Return a Promise to the result of evaluating the async `fn` with `...args` or undefined if
+ * evaluation fails.
+ */
+export const withErrorIgnoring = fn => async (...args) => {
   try {
     return await fn(...args)
   } catch (error) {
-    reportError(title, error)
+    // ignore error
   }
-})
+}
 
-// Transforms an async function so that it catches and ignores errors
-export const withErrorIgnoring = fn => async (...args) => {
+/**
+ * Return a Promise to the result of evaluating the async `fn` with `...args`. If evaluation fails,
+ * report the error to the user with `title` as a side effect.
+ */
+export const reportErrorAndRethrow = _.curry((title, fn) => async (...args) => {
   try {
     return await fn(...args)
   } catch (error) {
-    // ignore error
+    reportError(title, error)
+    throw error
   }
-}
+})
+
+/**
+ * Return a Promise to the result of evaluating the async `fn` with `...args` or undefined if
+ * evaluation fails. If evaluation fails, report the error to the user with `title`.
+ */
+export const withErrorReporting = _.curry((title, fn) => {
```

Comment:
As a point on style - I don't share the need to not have a new line after `=`, eg:
```javascript
export const withErrorReporting = 
  _.curry((title, fn) => withErrorIgnoring(reportErrorAndRethrow(title)(fn))
```